File: Replace every nth.py
def replace_nth(text, n, old_value, new_value):
    if n<=0:
        return text
    count = 0
    text_a = list(text)
    for i,s in enumerate(text):
        if s == old_value:
            count+=1
            if count%n==0:
                text_a[i]=new_value
    return "".join(text_a)
                
        
            
        

File: Replace every nth/Replace every nth.js
function replaceNth(text, n, oldValue, newValue) {
    if (n <= 0) {
        return text;
    }
    let count = 0;
    let textArray = text.split('');
    for (let i = 0; i < text.length; i++) {
        if (text[i] === oldValue) {
            count++;
            if (count % n === 0) {
                textArray[i] = newValue;
            }
        }
    }
    return textArray.join('');
}
